lib/Reader.php

```diff
@@ -322,4 +328,35 @@ function getDeserializerForElementName($name) {
 
     }
 
+    /**
+     * Set the URI containing the XML to parse
+     * @param  string  $uri
+     * @param  string  $encoding
+     * @param  int $options
+     * @return bool            Returns TRUE on success or FALSE on failure
+     */
+    function open($uri, $encoding = null, $options = 0) {
```

Comment:
I think the true end-goal of your patch is not track when `open` is called, because that would be much simpler. I think what you want is being able to stop `parseInnerTree()`. Therefore I think you should also just set `$this->stopped = false` in that function, not in `open`/`close`.